doc/nrf/libraries/caf/sensor_sampler.rst

```diff
@@ -169,6 +172,10 @@ The |sensor_sampler| samples sensors periodically, according to the configuratio
 Sampling of the sensors is done from a dedicated preemptive thread.
 You can change the thread priority by setting the :kconfig:`CONFIG_CAF_SENSOR_SAMPLER_THREAD_PRIORITY` Kconfig option.
 Use the preemptive thread priority to make sure that the thread does not block other operations in the system.
+The sensor sampler module limits number of :c:struct:sensor_event for each sensor that are submitted, but not processed to prevent OOM error in case system workqueue is blocked.
+The limit value for maximum unprocessed events for each sensor is placed in ``sensor_config.active_events_limit`` in configuration file.
+Counter ``active_sensor_events_cnt`` is incremented when sending :c:struct:sensor_event and decremented while event is processed by the sensor sampler that is a final subscriber of the event.
+Because of that the max number of allocated sensor events for given sensor equals active_events_limit + 1.
```

Comment:
```suggestion

For each sensor, the |sensor_sampler| limits the number of ``sensor_event`` events that it submits, but whose processing has not been completed.
This is done to prevent out-of-memory error if the system workqueue is blocked.
The limit value for the maximum number of unprocessed events for each sensor is placed in the ``sensor_config.active_events_limit`` structure field in the configuration file.
The ``active_sensor_events_cnt`` counter is incremented when sending ``sensor_event`` is sent and decremented when the event is processed by the sensor sampler that is the final subscriber of the event.
Because of this behavior, the maximum number of allocated sensor events for the given sensor is equal to ``active_events_limit`` plus one.
```